test/e2e_node/swap_test.go

```diff
@@ -0,0 +1,254 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"fmt"
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/pkg/kubelet/types"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	testutils "k8s.io/kubernetes/test/utils"
+	admissionapi "k8s.io/pod-security-admission/api"
+	"path/filepath"
+	"strconv"
+)
+
+const (
+	cgroupBasePath        = "/sys/fs/cgroup/"
+	cgroupV1SwapLimitFile = "/memory/memory.memsw.limit_in_bytes"
+	cgroupV2SwapLimitFile = "memory.swap.max"
+	cgroupV1MemLimitFile  = "/memory/memory.limit_in_bytes"
+)
+
+var _ = SIGDescribe("Swap [NodeConformance]", func() {
+	f := framework.NewDefaultFramework("swap-test")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelBaseline
+
+	ginkgo.DescribeTable("with configuration", func(qosClass v1.PodQOSClass, memoryRequestEqualLimit bool) {
+		ginkgo.By(fmt.Sprintf("Creating a pod of QOS class %s. memoryRequestEqualLimit: %t", qosClass, memoryRequestEqualLimit))
+		pod := getSwapTestPod(f, qosClass, memoryRequestEqualLimit)
+		pod = runPodAndWaitUntilScheduled(f, pod)
+
+		isCgroupV2 := isPodCgroupV2(f, pod)
+		isLimitedSwap := isLimitedSwap(f, pod)
+
+		if !isSwapFeatureGateEnabled() || !isCgroupV2 || (isLimitedSwap && (qosClass != v1.PodQOSBurstable || memoryRequestEqualLimit)) {
```

Comment:
I don't think the feature gate check will work as you expect in `e2e_node` tests. We had many issues with this as feature gates might not be loaded properly. Can you please double check on this

Reply:
Interesting. I see this being used in other e2e tests, e.g. here: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apimachinery/chunking.go#L42.

WDYT? @harche @haircommander

Reply:
I think you can use [withFeatureGate](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/util.go#L525) instead

Reply:
> I think you can use [withFeatureGate](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/util.go#L525) instead

Thanks @haircommander, but `withFeatureGate` actually [uses](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/util.go#L526) `utilfeature.DefaultFeatureGate.Enabled()`.

It's also used in other places under `test/`, e.g. [here](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apimachinery/chunking.go#L42) and [here](https://github.com/kubernetes/kubernetes/blob/master/test/integration/apiserver/apiserver_test.go#L579).

Are we sure there's something wrong with it?